File: test-db.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const dbPath = path.join(__dirname, 'database', 'team_directory.db');

console.log('Testing database connection...');
console.log(`Database path: ${dbPath}`);

const db = new sqlite3.Database(dbPath, (err) => {
    if (err) {
        console.error('❌ Error connecting to database:', err.message);
        process.exit(1);
    }
    console.log('✅ Connected to database');
});

db.all('SELECT COUNT(*) as count FROM Employees', (err, rows) => {
    if (err) {
        console.error('❌ Error querying Employees:', err.message);
    } else {
        console.log(`✅ Employee count: ${rows[0].count}`);
        
        if (rows[0].count === 0) {
            console.log('⚠️  Database is empty! Initializing with sample data...');
            initializeDatabase();
        } else {
            console.log('\n✅ Sample employees:');
            db.all('SELECT * FROM Employees LIMIT 5', (err, employees) => {
                if (err) {
                    console.error(err);
                } else {
                    console.table(employees);
                }
                db.close();
            });
        }
    }
});

function initializeDatabase() {
    const sampleEmployees = [
        { FirstName: 'Ahmed', LastName: 'Khan', Role: 'Software Engineer' },
        { FirstName: 'Fatima', LastName: 'Ali', Role: 'Product Manager' },
        { FirstName: 'Hassan', LastName: 'Malik', Role: 'UX Designer' },
        { FirstName: 'Ayesha', LastName: 'Raza', Role: 'Data Analyst' },
        { FirstName: 'Usman', LastName: 'Sheikh', Role: 'DevOps Engineer' },
        { FirstName: 'Zainab', LastName: 'Iqbal', Role: 'Frontend Developer' },
        { FirstName: 'Bilal', LastName: 'Hussain', Role: 'Backend Developer' },
        { FirstName: 'Sana', LastName: 'Ahmed', Role: 'QA Engineer' }
    ];

    const stmt = db.prepare('INSERT INTO Employees (FirstName, LastName, Role) VALUES (?, ?, ?)');
    
    sampleEmployees.forEach((employee) => {
        stmt.run(employee.FirstName, employee.LastName, employee.Role, (err) => {
            if (err) console.error('Error inserting:', err.message);
        });
    });

    stmt.finalize(() => {
        console.log(`✅ Inserted ${sampleEmployees.length} employee records`);
        
        db.all('SELECT * FROM Employees', (err, employees) => {
            if (err) {
                console.error(err);
            } else {
                console.log('\n✅ Database initialized! Current records:');
                console.table(employees);
            }
            db.close();
        });
    });
}

File: test-db.py
import sqlite3
import os
from pathlib import Path

db_path = Path(__file__).parent / 'database' / 'team_directory.db'

print(f"Database path: {db_path}")
print(f"Database exists: {db_path.exists()}")

if not db_path.exists():
    print("❌ Database file not found!")
    exit(1)

conn = sqlite3.connect(str(db_path))
cursor = conn.cursor()

try:
    cursor.execute("SELECT COUNT(*) FROM Employees")
    count = cursor.fetchone()[0]
    print(f"✅ Current employee count: {count}")
    
    if count == 0:
        print("\n⚠️  Database is empty. Inserting sample data...")
        
        employees = [
            ('Ahmed', 'Khan', 'Software Engineer'),
            ('Fatima', 'Ali', 'Product Manager'),
            ('Hassan', 'Malik', 'UX Designer'),
            ('Ayesha', 'Raza', 'Data Analyst'),
            ('Usman', 'Sheikh', 'DevOps Engineer'),
            ('Zainab', 'Iqbal', 'Frontend Developer'),
            ('Bilal', 'Hussain', 'Backend Developer'),
            ('Sana', 'Ahmed', 'QA Engineer')
        ]
        
        cursor.executemany(
            'INSERT INTO Employees (FirstName, LastName, Role) VALUES (?, ?, ?)',
            employees
        )
        conn.commit()
        print(f"✅ Inserted {len(employees)} employee records")
    
    print("\n📋 Database contents:")
    cursor.execute("SELECT ID, FirstName, LastName, Role FROM Employees ORDER BY LastName")
    rows = cursor.fetchall()
    for row in rows:
        print(f"  {row[0]}: {row[1]} {row[2]} - {row[3]}")
        
except Exception as e:
    print(f"❌ Error: {e}")
finally:
    conn.close()
